rkt/image/httpops.go

```diff
@@ -91,19 +91,19 @@ func (o *httpOps) DownloadImageWithETag(u *url.URL, etag string) (readSeekCloser
 	if err != nil {
 		return nil, nil, err
 	}
-	defer func() { maybeClose(aciFile) }()
 
 	session := o.getSession(u, aciFile.File, "ACI", etag)
 	dl := o.getDownloader(session)
 	if err := dl.Download(u, aciFile.File); err != nil {
+		aciFile.Close() // download failed, close the aciFile to implicitely release the lock
```

Comment:
Aha, there you can see why I added this `maybeClose` function… To avoid closing the file in every error case. Instead I preferred the deffered `maybeClose` function call that does nothing if `aciFile` became `nil` (which would happen if no errors happened).